src/types/stack.rs

```diff
@@ -125,3 +194,10 @@ impl<'a> StackArg<'a> {
         &mut self.0
     }
 }
+
+/// Snapshot of a stack frame state.
+pub struct StackState {
```

Comment:
should probably contain the current param as well, because it could be called when the current param is non-zero